sycli/src/main.rs

```diff
@@ -665,3 +667,120 @@ fn main() {
         _ => {}
     }
 }
+
+/// Parse search criteria out of a filter string
+fn parse_filter(searches: &str) -> Vec<Criterion> {
+    use rpc::criterion::{Operation, Value};
+
+    let mut criterion = Vec::new();
+
+    // possible delimiters
+    let delimiters = vec![">=", "<=", "==", "!=", "<", ">", ":"];
+
+    'search: for search in searches.split_whitespace() {
+        for delim in &delimiters {
+            if search.split(delim).count() == 2 {
+                let field = search.split(delim).nth(0).unwrap().to_string();
+                let value = search.split(delim).nth(1).unwrap();
+
+                let value = if field == "name"
```

Comment:
I think it's a good idea to do a match here and be explicit about possible field names we could expect. Then we can print to stderr that an unrecognized field name was used if nothing matches.